src/mono/msbuild/AotCompilerTask/Utils.cs

```diff
@@ -0,0 +1,88 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+internal class Utils
+{
+    public static string RunProcess(
+        string path,
+        string args = "",
+        IDictionary<string, string>? envVars = null,
+        string? workingDir = null,
+        bool ignoreErrors = false)
+    {
+        LogInfo($"Running: {path} {args}");
+        var outputBuilder = new StringBuilder();
+        var errorBuilder = new StringBuilder();
+        var processStartInfo = new ProcessStartInfo
+        {
+            FileName = path,
+            UseShellExecute = false,
+            CreateNoWindow = true,
+            RedirectStandardError = true,
+            RedirectStandardOutput = true,
+            Arguments = args,
```

Comment:
Depending on the platform you're running on, you might be able to provide a list of arguments to the [`ArgumentList`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.processstartinfo.argumentlist?view=netcore-3.1) property instead (in which case you won't have to worry about quoting).

Reply:
I should've mentioned in the PR description that this Utils code won't go upstream to dotnet/sdk so please ignore it. They have Process launching helpers there already :)